src/io/fastx.rs

```diff
@@ -0,0 +1,548 @@
+    error_has_occured: bool,
+}
+
+impl Record {
+    pub fn new() -> Self {
+        Record::FASTQ(fastq::Record::new())
+    }
+
+    pub fn with_attrs(
+        id: &str,
+        desc: Option<&str>,
+        seq: TextSlice<'_>,
+        qual: Option<&[u8]>,
+    ) -> Self {
+        match qual {
+            Some(q) => Record::FASTQ(fastq::Record::with_attrs(id, desc, seq, q)),
+            None => Record::FASTA(fasta::Record::with_attrs(id, desc, seq)),
+        }
+    }
+
+    pub fn id(&self) -> &str {
+        match self {
+            Record::FASTA(x) => x.id(),
+            Record::FASTQ(x) => x.id(),
+        }
+    }
+
+    pub fn check(&self) -> Result<(), &str> {
+        match self {
+            Record::FASTA(x) => x.check(),
+            Record::FASTQ(x) => x.check(),
+        }
+    }
+
+    pub fn desc(&self) -> Option<&str> {
+        match self {
+            Record::FASTA(x) => x.desc(),
+            Record::FASTQ(x) => x.desc(),
+        }
+    }
+
+    pub fn seq(&self) -> TextSlice<'_> {
+        match self {
+            Record::FASTA(x) => x.seq(),
+            Record::FASTQ(x) => x.seq(),
+        }
+    }
+
+    pub fn qual(&self) -> Option<&[u8]> {
+        match self {
+            Record::FASTA(_) => None,
+            Record::FASTQ(x) => Some(x.qual()),
+        }
+    }
+
+    pub fn set_base_qual(&mut self, qual: &[u8]) {
+        let new_record = fastq::Record::with_attrs(self.id(), self.desc(), self.seq(), qual);
+        mem::replace(self, Record::FASTQ(new_record));
+    }
+}
+
+impl<R> fasta::FastaRead for Reader<R>
+where
+    R: io::Read,
+{
+    fn read(&mut self, record: &mut fasta::Record) -> io::Result<()> {
+        if let Reader::Uninitialized(_) = self {
+            self.init_reader()?;
+        };
+
+        match self {
+            Reader::FASTA(x) => x.read(record),
+            Reader::FASTQ(x) => {
+                let mut fq_record = fastq::Record::new();
+                x.read(&mut fq_record)?;
+                mem::replace(
+                    record,
+                    fasta::Record::with_attrs(fq_record.id(), fq_record.desc(), fq_record.seq()),
+                );
+                Ok(())
+            }
+            Reader::Uninitialized(_) => unreachable!(),
+        }
+    }
+}
+
+impl<R> fastq::FastqRead for Reader<R>
+where
+    R: io::Read,
+{
+    fn read(&mut self, record: &mut fastq::Record) -> io::Result<()> {
+        if let Reader::Uninitialized(_) = self {
+            self.init_reader()?;
+        };
+
+        match self {
+            Reader::FASTA(x) => {
+                let mut fa_record = fasta::Record::new();
+                x.read(&mut fa_record)?;
+                mem::replace(
+                    record,
+                    fastq::Record::with_attrs(
+                        fa_record.id(),
+                        fa_record.desc(),
+                        fa_record.seq(),
+                        &vec![BASE_QUAL; fa_record.seq().len()],
```

Comment:
Do we want to have it possible to use a `Reader::FASTA(x)` to generate a `fastq::Record` with dummy `BASE_QUAL`s? I would think that the idea to enable explicitly upgrading a `fasta::Record` to a `fastq::Record` by providing a `qual` string is much safer (see line 114 above, `pub fn set_base_qual()`). Also, doesn't this implementation contradict the `impl<R> FastqRead for Reader<R>`  of `read()` in `src/io/fastq.rs`, especially the new `Err()` in line 149? But maybe I'm missing something about the code structure, here?